generics.go

```diff
@@ -0,0 +1,24 @@
+//go:build go1.18
+
+package container
+
+func ResolveT[T any](c Container) (T, error) {
```

Comment:
note that it's not possible to have generic receiver methods in go 1.18.

i.e. it's not possible to have `func (c Container) ResolveT[T any]() (T, error)` because methods cannot have type parameters.

this is partly why the new `Must` methods i've added all take the container as an argument; for future consistency with generic methods.

Reply:
Would you consider making a /v4 that converted the existing default container methods like `container.Singleton(resolver)` to `container.Singleton(container, resolver)`?

Could then export `var container` as `var Default` so that users can update and call `container.Singleton(container.Default, resolver)`